core/cc/archive.cpp

```diff
@@ -210,6 +210,10 @@ Archive::Archive(const std::string& archiveName)
     GAPID_FATAL("Unable to open archive index file %s", indexFilename.c_str());
   }
 
+  // Linux fopen() with mode "a" leads to reads from the beginning of file, but
+  // this is not true on Android, hence the explicit rewind() here
+  rewind(mIndexFile);
```

Comment:
This seems counter to what people will expect, we may want to file a bug.

Reply:
I filed a bug internally.